### PR TITLE
Premigration checklist for OCP 4 clusters

### PR DESCRIPTION
**Pull request by apinnick**
No BZ or Jira ID

Update migration checklist for MTC, based on the [OCP3>4 checklists](https://docs.openshift.com/container-platform/4.7/migrating_from_ocp_3_to_4/premigration-checklists.html).

4.5+

Preview: https://deploy-preview-33562--osdocs.netlify.app/openshift-enterprise/latest/migration-toolkit-for-containers/premigration-checklists.html

QE approved.

### CODE REVIEW

**Comment by apinnick** [comment]
@XinRedhat  Can you review this? Just need to make sure that the checklist items are suitable for migrating between OCP 4 clusters and that I haven't left anything out by accident when I adapted the OCP3>4 checklists.

**Review by sergiordlr — APPROVED** [review]
Looks good to me.

Thank you very much!

File: migration-toolkit-for-containers/premigration-checklists.adoc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+* [ ] The registry uses a link:../scalability_and_performance/optimizing-storage.html#recommended-configurable-storage-technology_persistent-storage[recommended storage type].
+* [ ] You can read and write images to the registry.
+* [ ] The link:https://access.redhat.com/articles/3093761[etcd cluster] is healthy.
+* [ ] The xref:../post_installation_configuration/node-tasks.adoc#create-a-kubeletconfig-crd-to-edit-kubelet-parameters_post-install-node-tasks[average API server response time] on the source cluster is less than 50 ms.
+
+[id="target-cluster-checklist_{context}"]
+== Target cluster
+
+* [ ] The cluster has xref:../storage/dynamic-provisioning.adoc#defining-storage-classes_dynamic-provisioning[storage classes] defined for the storage types used by the source cluster, for example, block volume, file system, or object storage.
```
[the remark this answers is not in the document]
Comment (XinRedhat) [reply migration-toolkit-for-containers/premigration-checklists.adoc:72]: other looks good

[the remark this answers is not in the document]
Comment (apinnick) [reply migration-toolkit-for-containers/premigration-checklists.adoc:72]: Thanks. Wasn't sure about this one.

File: migration-toolkit-for-containers/premigration-checklists.adoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+[id="cluster-health-checklist_{context}"]
+== Cluster health checklist
+
+* [ ] The clusters meet the minimum hardware requirements for the specific platform and installation method, for example, on xref:../installing/installing_bare_metal/installing-bare-metal.adoc#minimum-resource-requirements_installing-bare-metal[bare metal].
+* [ ] The {mtc-short} version is the same on all clusters.
+* [ ] All xref:../migration-toolkit-for-containers/migrating-applications-with-mtc.adoc#migration-prerequisites_migrating-applications-with-mtc[{mtc-short} prerequisites] are met.
+* [ ] All nodes have an active {product-title} subscription.
+* [ ] You have xref:../support/troubleshooting/verifying-node-health.adoc#verifying-node-health[verified node health].
+* [ ] The link:../authentication/understanding-identity-provider.adoc#supported-identity-providers[identity provider] is working.
```
Comment (ahardin-rh) [inline migration-toolkit-for-containers/premigration-checklists.adoc:18]: s/link:/xref:  
The link is currently failing

File: migration-toolkit-for-containers/premigration-checklists.adoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* [ ] The clusters have sufficient resources for migration.
++
+[NOTE]
+====
+Clusters require additional memory, CPUs, and storage in order to run a migration on top of normal workloads. Actual resource requirements depend on the number of Kubernetes resources being migrated in a single migration plan. You must test migrations in a non-production environment in order to estimate the resource requirements.
+====
+
+* [ ] The link:https://access.redhat.com/solutions/4885641[etcd disk performance] of the clusters has been checked with `fio`.
+
+== Source cluster
```
Comment (ahardin-rh) [inline migration-toolkit-for-containers/premigration-checklists.adoc:29]: [id="source-cluster-checklist_{context}"]

**Comment by ahardin-rh** [comment]
@apinnick Just a few minor comments from me. Thanks!

**Pull request merged**